script.js

```diff
@@ -1,106 +1,112 @@
-        return;
+    const taskName = todoInput.value;
+    if (taskName === "") {
+        alert(ERROR);
+    } else {
+        addTask(taskName);
     }
+})
+
 
-    addTask(newTaskName);
-    TODO_INPUT.value = "";
-});
 
 function addTask(taskName, isCompleted = false) {
     const listItem = document.createElement("li");
     listItem.classList.toggle(TASK_COMPLETED_CLASS, isCompleted);
 
     const taskNameSpan = document.createElement("span");
     taskNameSpan.textContent = taskName;
-    taskNameSpan.style.textDecoration = isCompleted ? "line-through" : "none";
     listItem.appendChild(taskNameSpan);
+    taskNameSpan.style.textDecoration = isCompleted ? "line-through" : "none";
 
-    const isTaskDoneCheckbox = document.createElement("input");
-    isTaskDoneCheckbox.setAttribute("type", "checkbox");
-    isTaskDoneCheckbox.checked = isCompleted;
-    listItem.appendChild(isTaskDoneCheckbox);
+    const taskCheckbox = document.createElement("input");
+    taskCheckbox.setAttribute("type", "checkbox");
+    isCompleted = taskCheckbox.checked;
+    listItem.appendChild(taskCheckbox);
 
     const deleteButton = document.createElement("button");
-    deleteButton.textContent = DELETE_BUTTON_LABEL;
+    deleteButton.textContent = DELETE_BUTTON;
     listItem.appendChild(deleteButton);
 
-    const editButton = document.createElement("button");
-    editButton.textContent = EDIT_BUTTON_LABEL;
-    listItem.appendChild(editButton);
+    const editButtton = document.createElement("button");
 
-    TODO_LIST.appendChild(listItem);
+    editButtton.textContent = EDIT_BUTTON;
+    listItem.appendChild(editButtton);
 
-    isTaskDoneCheckbox.addEventListener("change", function () {
-        taskNameSpan.style.textDecoration = this.checked ? "line-through" : "none";
+    todoList.appendChild(listItem);
+    todoInput.value = "";
 
+    taskCheckbox.addEventListener("change", function () {
+        taskNameSpan.style.textDecoration = this.checked ? "line-through" : "none";
         listItem.classList.toggle(TASK_COMPLETED_CLASS, this.checked);
 
-        saveTasksToLocalStorage();
-    });
+        saveToLocalStorage();
+    })
 
     deleteButton.addEventListener("click", function () {
-        TODO_LIST.removeChild(listItem);
+        todoList.removeChild(listItem);
 
-        saveTasksToLocalStorage();
-    });
- 
-    const taskNameEditInput = document.createElement("input");
-    taskNameEditInput.type = "text";
-    editButton.addEventListener("click", function () {
+        saveToLocalStorage();
+    })
+
+    const taskEditingInput = document.createElement("input");
+
+    editButtton.addEventListener("click", function() {
         const isEditing = listItem.classList.contains(TASK_EDITING_CLASS);
 
         if (isEditing) {
-            taskNameSpan.textContent = taskNameEditInput.value;
-            listItem.replaceChild(taskNameSpan, taskNameEditInput);
+            taskNameSpan.textContent = this.previousSibling.value;
```

Comment:
As far as I see this operation is actually redundant, explain why.

Reply:
Changed to taskEditingInput.value.
I see why it's redundant. I replaced method insertBefore with replaceChild, because in this case we don't need to delete node manually.